### PR TITLE
Changes in License applyed

### PR DESCRIPTION
**Pull request by MaxThomasHPI**
- as discussed, there were problems/inconsistencies in the License object
- updated to the latest version
- documentation was aligned -> description attribute in the schema updated
- removed outdated uml diagrams
- please check!

### CODE REVIEW

**Comment by MrSerth** [comment]
Thanks for working on these changes. I'll leave it to @Dome-GER and @cwillems to have a detailed look. Just two remarks ahead:

- You've renamed the JSON schema from `moochub-schema.json` to `mooch_schema.json`. Is this change intended? Currently, Git tracks these two files as unrelated, meaning that one file is deleted and another (new one) is added. Hence, comparing the differences is very tedious on GitHub.
- You've saved the JSON schema, just converted from JSON syntax to YAML syntax, in the `moochub-api.v3.yml`. Therewith, the file no longer describes the API (as suggested by the name), but "just" the format of the response body. Hence, it means the `moochub-api.v3.yml` is becoming a real duplicate of the `mooch_schema.json` without added value.


PS: Feel free to adjust the commit message [especially the second one] to provide a better overview of the changes. [Here](https://cbea.ms/git-commit/) are some thoughts on writing good commit messages.

File: moochub-api.v3.yml [hunk 1]
```diff
@@ -1,1786 +1,1144 @@
...
+    properties:
+      attributes:
+        description: The Attributes of the course.
+        properties:
+          contributor:
+            description: A List of partner institutes or persons contributed to this
+              course.
+            items:
+              anyOf:
+              - &id001
```
Comment (Dome-GER) [inline moochub-api.v3.yml:16]: Looks like the indentation for array members is broken in this file.

**Review by Dome-GER — COMMENTED** [review]
Super hard to review. I think this needs a review of the new files without comparison.
Or what is your recommendation?

Why is the general schema quite different?

**Comment by MrSerth** [comment]
Thanks, it's getting easier to compare the changes. Still, I have a few remarks:

- Would you mind using the same indention level in the JSON file and the same representation for strings (with or without single quotes) in the YAML file or extract these changes to another commit / PR? Currently, those changes are all highlighted (as seen in the diff), which makes it more difficult to compare.
- Further, I noticed that the `$schema` reference given in the JSON might not be standard-compliant. Could you please double-check that?

**Comment by MaxThomasHPI** [comment]
have just recently found another error in my pull request, will take a look at these new comments afterward. 

speaking of `$schema`: is this about the missing "#" at the end of the link as discussed?

**Comment by MrSerth** [comment]
> speaking of $schema: is this about the missing "#" at the end of the link as discussed?

Potentially, yes. However, it seems to depend on the specific schema version used; the newer ones do not include the `#` any longer. Hence, I really suggest double-checking with the upstream documentation. [This page](https://json-schema.org/specification-links) contains the actual specifications, linking to the respective "JSON Schema meta-schema" for each version. Besides the `#`, there are differences in the protocol used (`http` vs `https`) depending on the version.

**Comment by MrSerth** [comment]
Just another remark: All URLs pointing to <https://openhpi.azureedge.net> are no longer valid, since we are stopped using this host to deliver assets.

**Comment by MaxThomasHPI** [comment]
ok, is this relevant? Should only be an issue with the images, and they are just examples? Could also e something like: www.example.com/image.png?

**Comment by MrSerth** [comment]
> ok, is this relevant?

Well, it doesn't make the example easier to understand nor complete. Maybe it's just something for a dedicated ticket / PR? The smaller each ticket and the corresponding PR is, the easier (and faster!) they are to review 😉.

> [...] they are just examples? 

Yes, and they serve the purpose to illustrate with valid examples how an entry could look like (at least that's how I would argue).

File: moochub-schema.json [hunk 2]
```diff
@@ -1,3177 +1,3188 @@
...
+                                            },
+                                            "url": {
+                                                "type": [
+                                                    "string",
+                                                    "null"
+                                                ],
+                                                "format": "iri",
+                                                "description": "A license according to https://spdx.org/licenses/ or \"null\" if proprietary.",
+                                                "example": "https://creativecommons.org/publicdomain/zero/1.0"
+                                            },
```
Comment (MrSerth) [inline moochub-schema.json:256]: Here, for example, I see another issue: The example given doesn't seem to be consistent. 

- The `identifier` is given as `CC-BY-SA-4.0`, but the `url` is `https://creativecommons.org/publicdomain/zero/1.0`.
- The `url` is syntactically valid, but doesn't match to the "description", where a license in accordance with <https://spdx.org/licenses/> is requested.

**Comment by MaxThomasHPI** [comment]
> > speaking of $schema: is this about the missing "#" at the end of the link as discussed?
> 
> Potentially, yes. However, it seems to depend on the specific schema version used; the newer ones do not include the `#` any longer. Hence, I really suggest double-checking with the upstream documentation. [This page](https://json-schema.org/specification-links) contains the actual specifications, linking to the respective "JSON Schema meta-schema" for each version. Besides the `#`, there are differences in the protocol used (`http` vs `https`) depending on the version.

It seems like draft 7 has the `#` at the end and uses http. Schould we go for the latest version 2020-12? in this case, there is no `#` and the protocoll would be https...

**Comment by MrSerth** [comment]
> > > speaking of $schema: is this about the missing "#" at the end of the link as discussed?
> > 
> > 
> > Potentially, yes. However, it seems to depend on the specific schema version used; the newer ones do not include the `#` any longer. Hence, I really suggest double-checking with the upstream documentation. [This page](https://json-schema.org/specification-links) contains the actual specifications, linking to the respective "JSON Schema meta-schema" for each version. Besides the `#`, there are differences in the protocol used (`http` vs `https`) depending on the version.
> 
> It seems like draft 7 has the `#` at the end and uses http. Schould we go for the latest version 2020-12? in this case, there is no `#` and the protocoll would be https...

I don’t mind, feel free to use the latest version. In that case, however, you should also check for potential changes.

File: moochub-schema.json (hunk 2 above)
Comment (MaxThomasHPI) [reply moochub-schema.json]: fixed

**Pull request merged**